### ISSUE TITLE
Not all pending takes on stream are resolved when stream is closed

### ISSUE THREAD

**Issue by dparis** [ztellman/manifold#27]
Here's what I'm seeing:

```clojure
(def s (ms/stream))

(dotimes
 [n 10]
 (doto
  (Thread.
   (fn
    []
    (loop
     []
     (when-let
      [x @(ms/take! s)]
      (recur)))
    (println (str "Thread" n "ended"))))
  (.start)))

s  ;; << stream: {:pending-puts 0, :drained? false, :buffer-size 0, 
   ;;             :permanent? false, :type "manifold", :sink? true, :closed? false, 
   ;;             :pending-takes 10, :buffer-capacity 0, :source? true} >>

(ms/close! s)  ;; "Thread 0 ended"

s  ;; << stream: {:pending-puts 0, :drained? true, :buffer-size 0, 
   ;;             :permanent? false, :type "manifold", :sink? true, :closed? false, 
   ;;             :pending-takes 9, :buffer-capacity 10, :source? true} >>
```

Since all 10 threads are blocking on dereferencing the deferred returned by `take!`, I would expect that once `close!` is called on the stream, they'd all get `nil` and each would end. Instead, only one pending take seems to resolve, and the rest are kind of spinlocked.

Bug? Or am I making an incorrect assumption about the way `close!` works with the `take!` function? Would the `drained?` fix you made affect this? I haven't updated to the snapshot build yet, but I could if that might make a difference.

**Comment by dparis** [comment]
Got the fix for this, should have a PR submitted shortly.

**Comment by ztellman** [comment]
Definitely a bug, thanks for catching it.

**Comment by dparis** [comment]
Fixed by #29
